### PR TITLE
Add migration to add create_at column in external_service_repo table

### LINKED ISSUE #27795: Log an additional event every time a duplicate repo is added
Details in this slack thread: https://sourcegraph.slack.com/archives/CHPC7UX16/p1626822437340100

TL;DR; the table (public.repo) with data on all the repos added to cloud (including the date they were added) does not contain any duplicates (details on tables in this doc). So if User A adds Repo1 on 2021-07-15 and User B also adds Repo1, but three days later on 2021-07-18, Repo1 shows up once in the public.repo table, with a created date of 2021-07-15.

**What this means:**
* Continuing the above scenario: the data will show that User A and User B both added Repo1 to their sourcegraph account on 2021-07-15
* TL;DR any “repos added/deleted over time” analysis will be inaccurate using existing data, for any repo that was added more than once

### PR DESCRIPTION
**Pull request by varsanojidan**
Closes #27795

Adding an additional row to the external_service_repos table to log when repos are added by users so we can keep track of the added times for the repos if multiple adds are made.

Example:
image

## Test plan
Tested by running migration up and down against development environment, as well as adding a new repo and verifying the default created_at time was set correctly in the DB table

### CODE REVIEW

**Comment by varsanojidan** [comment]
@efritz Hi Eric, I was referred to you by Ryan to double check that making this migration on a large table like external_service_repos wouldn't introduce any unexpected issues, do you think it should be alright?

File: migrations/frontend/1646153853/up.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+BEGIN;
+
+ALTER TABLE IF EXISTS external_service_repos
+    ADD COLUMN IF NOT EXISTS created_at timestamp with time zone DEFAULT transaction_timestamp();
```
Comment (efritz) [inline migrations/frontend/1646153853/up.sql:4]: Thanks for reaching out!

At first I thought that adding a non-nullable column with a default might rewrite part of the table's heap space, but it looks like Postgres 11+ (we require Postgres 12+) can store non-volatile DEFAULT values in the catalogue for new columns.

This _should_ be a quick operation, and you shouldn't have to do any dances around adding a nullable column, adding a default, then making it non-nullable later.

LGTM!

File: migrations/frontend/1646153853/up.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+BEGIN;
```
Comment (efritz) [inline migrations/frontend/1646153853/up.sql:1]: FYI we wrap all migrations in transactions in the migration runner, so this isn't necessary anymore.

**Review by efritz — APPROVED** [review]

File: migrations/frontend/1646153853/up.sql (hunk 2 above)
Comment (varsanojidan) [reply migrations/frontend/1646153853/up.sql:1]: Do you think its better to remove it in that case?

Comment (efritz) [reply migrations/frontend/1646153853/up.sql:1]: It shouldn't hurt anything but it might feel cleaner is all 🤷 

Not blocking but feel free to pre-merge.

**Comment by ryanslade** [comment]
@varsanojidan Something we forgot was to make the `created_at` field `non null` so I think we should add another migration for that.

**Comment by efritz** [comment]
> Something we forgot was to make the created_at field non null so I think we should add another migration for that.

@ryanslade You'll need to wait a release after it's backfilled, otherwise it's not a backwards compatible change.

**Comment by varsanojidan** [comment]
Would the backwards compatibility not be taken care of by us setting a default to the field in a previous migration @efritz ?

**Comment by efritz** [comment]
@varsanojidan Actually if the column was added with a default then insertions (in the previous release) should still work.

**Comment by varsanojidan** [comment]
Opened a PR for making the column not nullable here:
https://github.com/sourcegraph/sourcegraph/pull/32078

**Pull request merged**